Lib/test/test_pyrepl/test_pyrepl.py

```diff
@@ -920,6 +929,84 @@ def test_exposed_globals_in_repl(self):
 
         self.assertTrue(case1 or case2 or case3 or case4, output)
 
+    def _assertMatchOK(
+            self, var: str, expected: str | re.Pattern, actual: str
+    ) -> None:
+        if isinstance(expected, re.Pattern):
+            self.assertTrue(
+                expected.match(actual),
+                f"{var}={actual} does not match {expected.pattern}",
+            )
+        else:
+            self.assertEqual(
+                actual,
+                expected,
+                f"expected {var}={expected}, got {var}={actual}",
+            )
```

Comment:
Why not use [`unittest.TestCase.assertRegex`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRegex)?

Reply:
This one match is dynamic depending on what's in the dictionary so it has to support both strings and patterns. I guess I could make them *all* patterns but that makes the dictionaries more noisy. 

More importantly, in the `re.Pattern` case I did start with `assertRegex` first, but it uses `re.search` and not `re.match` so the behavior is different. The `actual` string is already matched from a previous regular expression so it can and should be matched _exactly_.

Reply:
I see. `assertRegex` supports regex strings as well as `re.Pattern` objects, so I'd be tempted to just emulate `re.fullmatch()` by using regex strings as the values for all the dictionaries, starting each regex string with `^`, and ending each regex string with `$`. But I don't have a strong opinion!

Reply:
I see what you're saying and there's definitely space for improvement there. Given that 3.13 beta 4 will b "any minute now", I'd rather not play test golf right now. What do you say we land this as is and iterate on the tests after?

Reply:
SGTM